Tests/JetworkingTests/ClientTests/ClientTests.swift

```diff
@@ -173,6 +189,53 @@ final class ClientTests: XCTestCase {
         waitForExpectations(timeout: 5.0, handler: nil)
     }
 
+    func testPostRequestWithRequestWithServerError() {
+        let client = Client(configuration: Configurations.default())
+
+        let expectation = self.expectation(description: "Wait for post with empty content")
+        client.post(endpoint: Endpoints.voidPostServerError) { response, result in
+            dispatchPrecondition(condition: .onQueue(DispatchQueue.main))
+
+            switch result {
+            case .failure(APIError.serverError(statusCode: 500, error: _)):
+                XCTAssertNotNil(response)
+                XCTAssertEqual(response?.statusCode, 500)
+                expectation.fulfill()
+
+            case .failure:
+                XCTFail("Request should not result in failure!")
+
+            case .success:
+                XCTFail("Request should not result in failure!")
+            }
+        }
+
+        waitForExpectations(timeout: 5.0, handler: nil)
+    }
+
+    func testPostRequestWithRequestWithClientError() {
+        let client = Client(configuration: Configurations.default())
+        let expectation = self.expectation(description: "Wait for post with empty content")
+        client.post(endpoint: Endpoints.voidPostClientError) { response, result in
+            dispatchPrecondition(condition: .onQueue(DispatchQueue.main))
+
+            switch result {
+            case .failure(APIError.clientError(statusCode: 403, error: _)):
+                XCTAssertNotNil(response)
+                XCTAssertEqual(response?.statusCode, 403)
+                expectation.fulfill()
+
+            case .failure:
+                XCTFail("Request should not result in failure!")
+
+            case .success:
+                XCTFail("Request should not result in failure!")
```

Comment:
Please use a more meaningful failure message e.g. "Response should fail with status code 403"